pallets/ajuna-board/src/dot4gravity.rs

```diff
@@ -0,0 +1,216 @@
+// Ajuna Node
+// Copyright (C) 2022 BlogaTech AG
+
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+
+// You should have received a copy of the GNU Affero General Public License
+// along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+use ajuna_common::{Finished, TurnBasedGame};
+use dot4gravity::{Coordinates, Game as Dot4Gravity, GameState, RandomBlocksGenerator, Side};
+use frame_support::pallet_prelude::*;
+
+#[derive(Clone, Eq, PartialEq, Encode, Decode, RuntimeDebugNoBound, TypeInfo, MaxEncodedLen)]
+pub enum Turn {
+	DropBomb(Coordinates),
+	DropStone((Side, u8)),
+}
+
+pub struct Game<Account>(PhantomData<Account>);
+impl<Account> TurnBasedGame for Game<Account>
+where
+	Account: Parameter + Into<u32> + From<u32>,
+{
+	type Turn = Turn;
+	type Player = Account;
+	type State = GameState;
+
+	fn init(players: &[Self::Player]) -> Option<Self::State> {
+		if let [player_1, player_2] = players {
+			Some(Dot4Gravity::new_game::<RandomBlocksGenerator>(
+				(player_1.to_owned()).into(),
+				(player_2.to_owned()).into(),
+			))
+		} else {
+			None
+		}
+	}
+
+	fn play_turn(
+		player: Self::Player,
+		state: Self::State,
+		turn: Self::Turn,
+	) -> Option<Self::State> {
+		let player: u32 = player.into();
+		match turn {
+			Turn::DropBomb(coords) => Dot4Gravity::drop_bomb(state, coords, player),
+			Turn::DropStone((side, pos)) => Dot4Gravity::drop_stone(state, player, side, pos),
+		}
+		.ok()
+	}
+
+	fn is_finished(state: &Self::State) -> Finished<Self::Player> {
+		match Dot4Gravity::check_winner_player(*state).winner {
+			Some(winner) => Finished::Winner(winner.into()),
+			None => Finished::No,
+		}
+	}
+}
+
+#[cfg(test)]
```

Comment:
Just added a separate runtime to integrate `dot4gravity::Game` (rather than replacing the tests we have for now).